idaes/generic_models/unit_models/heat_exchanger.py

```diff
@@ -233,16 +240,47 @@ def build(self):
         self._process_config()
         config = self.config
 
+        ########################################################################
+        # Add control volumes                                                  #
+        ########################################################################
+        hot_side = _make_heater_control_volume(
+            self,
+            config.hot_side_name,
+            config.hot_side_config,
+            dynamic=config.dynamic,
+            has_holdup=config.has_holdup,
+        )
+        cold_side = _make_heater_control_volume(
+            self,
+            config.cold_side_name,
+            config.cold_side_config,
+            dynamic=config.dynamic,
+            has_holdup=config.has_holdup,
+        )
+        # Add refernces to the hot side and cold side, so that we have solid
+        # names to refere to internally.  side_1 and side_2 also maintain
+        # compatability with older models.  Using add_object_reference keeps
+        # these from showing up when you iterate through pyomo compoents in a
+        # model, so only the user specified control volume names are "seen"
+        if not hasattr(self, "side_1"):
+            add_object_reference(self, "side_1", hot_side)
+        if not hasattr(self, "side_2"):
+            add_object_reference(self, "side_2", cold_side)
+        if not hasattr(self, "hot_side"):
+            add_object_reference(self, "hot_side", hot_side)
+        if not hasattr(self, "cold_side"):
+            add_object_reference(self, "cold_side", cold_side)
+
         ########################################################################
         # Add variables                                                        #
         ########################################################################
-        # Use side 1 units as basis
+        # Use hot side units as basis
         s1_metadata = config.hot_side_config.property_package.get_metadata()
 
         q_units = s1_metadata.get_derived_units("power")
         u_units = s1_metadata.get_derived_units("heat_transfer_coefficient")
         a_units = s1_metadata.get_derived_units("area")
-        t_units = s1_metadata.get_derived_units("temperature")
+        temp_units = s1_metadata.get_derived_units("temperature")
```

Comment:
I don't want to add more work, but it seems odd that this is `temp_units` whilst all the others are shorter now.

Reply:
This was changed to address @jghouse88's comment.  I think it's just longer due to potential confusion between time and temperature.